### DISCUSSION TITLE
Damage Types?

### DISCUSSION THREAD

**Discussion by KnightCaptainU7** [exult/exult#329] [General]
How are damage types used in Exult? Or even in the original?

There are these types listed:
Normal - most weapons
Fire - lit torch, fire sword, firedoom staff, etc
Magic - most magic weapons, lightning whip
Lightning - lightning wand, lightening spell
Ethereal - Unused in SI?
Sonic - Unused in SI?
Unnamed 6
Unnamed 7

Any ideas what the unnamed ones might be?

**Comment by marzojr** [comment]
Each type of damage has its own vulnerability flags (in monster data) and immunity (in both monster data and armor data). Unless specified, an attack gains the attacker's strength as bonus when dealing damage, and receives the target's armor as a penalty. Also unless specified, attacks must deal enough damage to destroy a powder keg before it detonates.

Normal is damage exactly that. It is the only damage type that can cause bleeding, and it is caused by falling, innate weapons, and calltrops.

When an enemy has a weapons (innate or otherwise, including ammo) that deal any form of damage other than normal, defeating that enemy gives bonus XP.

Fire damage is dealt by ammo and weapons that cause it, as well as fire fields and campfires. Any amount of fire damage will detonate a powder keg.

Magic damage is like normal damage that cannot cause bleeding.

Lightning damage ignores armor, does not give strength bonuses, and is painful -- any amount of it causes the screen to flash red when the avatar is hurt. Lightning damage cannot destroy objects. I think that Exult does not use this in all places it should; some investigations I have been doing indicate that this is also used for poison, hunger, and (in SI) cold damage in the original games. So maybe a better name might be in order.

Ethereal damage ignores armor and cannot destroy objects.

Sonic damage ignores armor. Exult uses this for poison, hunger, and (in SI) cold damage which, as I mentioned, may be wrong.

There is usecode in BG to deal magic and lightning damage. There is usecode in SI to deal normal, fire, magic, and lightning damage.

The unnamed ones are completely unused in BG and SI, as far as I can tell. They have no special effects in Exult other than granting bonus XP (as mentioned above) and having invulnerability and vulnerability flags for them. These things also hold true in the original games; I had tested them way back then by editing game files.

**Comment by KnightCaptainU7** [comment]
Lightning may be better named Shock.

Which ones include armor or ignore it?
Normal - Counts Armor
Fire - 
Magic - 
Lightning - Ignores Armor
Ethereal - Ignores Armor
Sonic - Ignores Armor
Unnamed 6 - 
Unnamed 7 - 

So if one wanted to add the Cap of Missile Protection, it would require repurposing 6 or 7 into "Missile damage" and then creating a cap that gives immunity to the same? And set the damage type on all missiles and bolts?

**Reply by marzojr** [reply]
Let me quote myself:

> *Unless specified, an attack* gains the attacker's strength as bonus when dealing damage, *and receives the target's armor as a penalty*. Also unless specified, attacks must deal enough damage to destroy a powder keg before it detonates.

Emphasis added. So all damage types are affected by armor except for lightning, sonic, and ethereal.
